rclrs/src/rcl_utils.rs

```diff
@@ -0,0 +1,61 @@
+use crate::rcl_bindings::*;
+use crate::{RclrsError, ToResult};
+
+use crate::RclrsError::IndexOutOfRange;
+use libc::c_void;
+use std::ptr::null_mut;
+
+/// Returns arguments type held by `rcl_arguments` basing on `rcl_get_count` and `rcl_get_indices` function pointers.
+///
+/// This function must be called after `rcl_arguments` was initialized. `args` must be array of input arguments passed to node/program.
+///
+/// SAFETY: `rcl_get_count` and `rcl_get_indices` has to be corresponding rcl API functions, e.g.:
+/// `rcl_arguments_get_count_unparsed` -> `rcl_arguments_get_unparsed`
+/// `rcl_arguments_get_count_unparsed_ros` -> `rcl_arguments_get_count_ros`
+/// ...
+pub(crate) fn get_rcl_arguments(
+    rcl_get_count: unsafe extern "C" fn(*const rcl_arguments_t) -> std::os::raw::c_int,
```

Comment:
Honestly I don't see how to wrap this common portions to separate functions. They're just too small and linked tightly to te rest. Maybe wrapping it into macro woud solve this, but it's a bit nasty and seems like overkill.
Current approach is on @nnmm request, so if there's still something wrong with it please agree on one version, since I'd like to start resolving other issues not being attached to this one for so long :/

Reply:
Ah, I had missed that. My apologies.

Reply:
Yes, that PR has gone on quite long, and I agree with @Nizerlak that no more common functions can be factored out.